pkg/server/cache.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/client-go/rest"
+
+	virtualcommandoptions "github.com/kcp-dev/kcp/cmd/virtual-workspaces/options"
+	cacheserver "github.com/kcp-dev/kcp/pkg/cache/server"
+	cacheoptions "github.com/kcp-dev/kcp/pkg/cache/server/options"
+)
+
+// installCacheServer integrates the cache-server into the kcp-server.
+// The new server is available at /services/cache/... HTTP path.
+func (s *Server) installCacheServer(ctx context.Context) error {
+	cacheServerOptions := cacheoptions.NewOptions(s.Options.Extra.RootDirectory)
```

Comment:
this looks like options/config/server skew.

Reply:
you were right, I think it looks better now, PTAL.